File: augment_reset.nim
#!/usr/bin/env nim
##[
Augment Extension Trial Reset Tool

This script resets the trial period for the Augment coding extension
by modifying the extension's configuration files.
Supports Windows, macOS, and Linux systems.

Main features:
- Automatically detects and closes running VS Code
- Backs up existing configuration
- Generates new random device ID
- Preserves user settings

Author: Based on cursor-reset by @triallord
Created: 2/Jun/2025
Converted to Nim
]##

import std/[os, json, times, random, strutils, strformat, osproc, terminal]
import std/[asyncdispatch]

proc waitForKeypress() {.async.} =
  when defined(windows):
    if getEnv("TERM") == "":
      echo "\nPress any key to exit..."
      discard getch()
  else:
    discard

proc isEditorRunning(): bool =
  try:
    when defined(windows):
      let output = execProcess("tasklist /FI \"IMAGENAME eq Code.exe\" /FI \"IMAGENAME eq Cursor.exe\"")
      return "code.exe" in output.toLower() or "cursor.exe" in output.toLower()
    elif defined(macosx):
      let output = execProcess("pgrep -x \"Code\" || pgrep -x \"Cursor\" || pgrep -x \"Code Helper\" || pgrep -x \"Cursor Helper\"")
      return output.len > 0
    elif defined(linux):
      let output = execProcess("pgrep -x \"code\" || pgrep -x \"cursor\" || pgrep -x \"Code\" || pgrep -x \"Cursor\"")
      return output.len > 0
    else:
      return false
  except:
    return false

proc killEditorProcess(): Future[bool] {.async.} =
  try:
    var command = ""
    when defined(windows):
      command = "taskkill /F /IM Code.exe /T & taskkill /F /IM Cursor.exe /T"
    elif defined(macosx):
      command = "pkill -9 \"Code\" & pkill -9 \"Cursor\""
    elif defined(linux):
      command = "pkill -9 \"code\" & pkill -9 \"cursor\""
    else:
      raise newException(OSError, "Unsupported operating system")

    discard execProcess(command)
    await sleepAsync(1500)
    
    if not isEditorRunning():
      return true
    return false
  except Exception as e:
    echo "Error closing editors: ", e.msg
    return false

proc formatTimestamp(date: DateTime): string =
  let year = $date.year
  let month = align($date.month.int, 2, '0')
  let day = align($date.monthday, 2, '0')
  let hours = align($date.hour, 2, '0')
  let minutes = align($date.minute, 2, '0')
  let seconds = align($date.second, 2, '0')
  let milliseconds = align($(date.nanosecond div 1_000_000), 3, '0')

  return fmt"{year}{month}{day}{hours}{minutes}{seconds}{milliseconds}"

proc backupFile(filePath: string): Future[string] {.async.} =
  try:
    let timestamp = formatTimestamp(now())
    let backupPath = fmt"{filePath}.{timestamp}.bak"
    copyFile(filePath, backupPath)
    return backupPath
  except Exception as e:
    raise newException(IOError, fmt"Backup failed: {e.msg}")

proc getAugmentConfigPaths(): seq[string] =
  let homedir = getHomeDir()
  var paths: seq[string] = @[]

  when defined(windows):
    let appdata = getEnv("APPDATA")
    let localappdata = getEnv("LOCALAPPDATA")
    
    # Main configuration paths
    paths.add(appdata / "Code" / "User" / "globalStorage" / "augment.augment" / "state.json")
    paths.add(appdata / "Cursor" / "User" / "globalStorage" / "augment.augment" / "state.json")
    paths.add(appdata / "Code" / "User" / "globalStorage" / "augment.augment" / "subscription.json")
    paths.add(appdata / "Cursor" / "User" / "globalStorage" / "augment.augment" / "subscription.json")
    paths.add(appdata / "Code" / "User" / "globalStorage" / "augment.augment" / "account.json")
    paths.add(appdata / "Cursor" / "User" / "globalStorage" / "augment.augment" / "account.json")
    
    # Additional cache and storage locations
    paths.add(appdata / "Code" / "Cache" / "augment.augment")
    paths.add(appdata / "Cursor" / "Cache" / "augment.augment")
    paths.add(appdata / "Code" / "CachedData" / "augment.augment")
    paths.add(appdata / "Cursor" / "CachedData" / "augment.augment")
    paths.add(localappdata / "Code" / "User" / "globalStorage" / "augment.augment")
    paths.add(localappdata / "Cursor" / "User" / "globalStorage" / "augment.augment")
    
  elif defined(macosx):
    # Main configuration paths
    paths.add(homedir / "Library" / "Application Support" / "Code" / "User" / "globalStorage" / "augment.augment" / "state.json")
    paths.add(homedir / "Library" / "Application Support" / "Cursor" / "User" / "globalStorage" / "augment.augment" / "state.json")
    paths.add(homedir / "Library" / "Application Support" / "Code" / "User" / "globalStorage" / "augment.augment" / "subscription.json")
    paths.add(homedir / "Library" / "Application Support" / "Cursor" / "User" / "globalStorage" / "augment.augment" / "subscription.json")
    paths.add(homedir / "Library" / "Application Support" / "Code" / "User" / "globalStorage" / "augment.augment" / "account.json")
    paths.add(homedir / "Library" / "Application Support" / "Cursor" / "User" / "globalStorage" / "augment.augment" / "account.json")
    
    # Additional cache and storage locations
    paths.add(homedir / "Library" / "Caches" / "Code" / "augment.augment")
    paths.add(homedir / "Library" / "Caches" / "Cursor" / "augment.augment")
    paths.add(homedir / "Library" / "Application Support" / "Code" / "Cache" / "augment.augment")
    paths.add(homedir / "Library" / "Application Support" / "Cursor" / "Cache" / "augment.augment")
    
  elif defined(linux):
    # Main configuration paths
    paths.add(homedir / ".config" / "Code" / "User" / "globalStorage" / "augment.augment" / "state.json")
    paths.add(homedir / ".config" / "Cursor" / "User" / "globalStorage" / "augment.augment" / "state.json")
    paths.add(homedir / ".config" / "Code" / "User" / "globalStorage" / "augment.augment" / "subscription.json")
    paths.add(homedir / ".config" / "Cursor" / "User" / "globalStorage" / "augment.augment" / "subscription.json")
    paths.add(homedir / ".config" / "Code" / "User" / "globalStorage" / "augment.augment" / "account.json")
    paths.add(homedir / ".config" / "Cursor" / "User" / "globalStorage" / "augment.augment" / "account.json")
    
    # Additional cache and storage locations
    paths.add(homedir / ".cache" / "Code" / "augment.augment")
    paths.add(homedir / ".cache" / "Cursor" / "augment.augment")
    paths.add(homedir / ".config" / "Code" / "Cache" / "augment.augment")
    paths.add(homedir / ".config" / "Cursor" / "Cache" / "augment.augment")
  else:
    raise newException(OSError, "Unsupported operating system")

  return paths

proc generateDeviceId(): string =
  randomize()
  var deviceId = ""
  for i in 0..<64:
    deviceId.add(toHex(rand(15), 1).toLower())
  return deviceId

proc generateEmail(): string =
  randomize()
  var randomString = ""
  for i in 0..<16:
    randomString.add(toHex(rand(15), 1).toLower())
  return fmt"user_{randomString}@example.com"

proc generateUserId(): string =
  randomize()
  var userId = ""
  for i in 0..<32:
    userId.add(toHex(rand(15), 1).toLower())
  return userId

proc getUserInput(prompt: string): Future[string] {.async.} =
  stdout.write(prompt)
  stdout.flushFile()
  return stdin.readLine()

proc resetAugmentTrial() {.async.} =
  try:
    echo "🔍 Checking for running editors..."
    if isEditorRunning():
      echo "⚠️ VS Code or Cursor is running, attempting to close..."
      if await killEditorProcess():
        echo "✅ Editors have been closed\n"
      else:
        echo "❌ Failed to close editors"
        return

    let configPaths = getAugmentConfigPaths()
    echo "📂 Found configuration paths: ", configPaths

    # Generate new account data
    echo "🎲 Generating new account data..."
    let newDeviceId = generateDeviceId()
    let newUserId = generateUserId()
    let userEmail = generateEmail()
    echo "✅ New account data generated successfully\n"

    # Calculate trial dates
    let trialStartDate = now()
    let trialEndDate = trialStartDate + 14.days

    for configPath in configPaths:
      echo fmt"\n🔄 Processing: {configPath}"

      try:
        # Create directory if it doesn't exist
        createDir(parentDir(configPath))

        # Backup existing config
        echo "💾 Backing up configuration..."
        try:
          let backupPath = await backupFile(configPath)
          echo fmt"✅ Configuration backup complete: {backupPath}\n"
        except:
          echo "ℹ️ No existing configuration to backup\n"

        # If it's a directory, remove it completely
        try:
          if dirExists(configPath):
            removeDir(configPath)
            echo "✅ Removed directory: " & configPath
            continue
        except:
          discard # Ignore errors if file/directory doesn't exist

        # Create new configuration based on file type
        var newConfig: JsonNode
        if "subscription.json" in configPath:
          newConfig = %*{
            "status": "active",
            "type": "trial",
            "startDate": trialStartDate.format("yyyy-MM-dd'T'HH:mm:ss'.'fff'Z'"),
            "endDate": trialEndDate.format("yyyy-MM-dd'T'HH:mm:ss'.'fff'Z'"),
            "isActive": true,
            "isExpired": false,
            "remainingDays": 14,
            "trialCount": 0,
            "lastTrialReset": nil,
            "previousTrials": newJArray()
          }
        elif "account.json" in configPath:
          newConfig = %*{
            "userId": newUserId,
            "email": userEmail,
            "deviceId": newDeviceId,
            "createdAt": trialStartDate.format("yyyy-MM-dd'T'HH:mm:ss'.'fff'Z'"),
            "lastLogin": trialStartDate.format("yyyy-MM-dd'T'HH:mm:ss'.'fff'Z'"),
            "isActive": true,
            "trialHistory": newJArray(),
            "deviceHistory": newJArray()
          }
        else:
          # Default state.json configuration
          newConfig = %*{
            "deviceId": newDeviceId,
            "userId": newUserId,
            "email": userEmail,
            "trialStartDate": trialStartDate.format("yyyy-MM-dd'T'HH:mm:ss'.'fff'Z'"),
            "trialEndDate": trialEndDate.format("yyyy-MM-dd'T'HH:mm:ss'.'fff'Z'"),
            "trialDuration": 14,
            "trialStatus": "active",
            "trialExpired": false,
            "trialRemainingDays": 14,
            "trialCount": 0,
            "usageCount": 0,
            "totalUsageCount": 0,
            "lastUsageDate": nil,
            "messageCount": 0,
            "totalMessageCount": 0,
            "lastReset": now().format("yyyy-MM-dd'T'HH:mm:ss'.'fff'Z'"),
            "firstRunDate": now().format("yyyy-MM-dd'T'HH:mm:ss'.'fff'Z'"),
            "lastRunDate": now().format("yyyy-MM-dd'T'HH:mm:ss'.'fff'Z'"),
            "sessionId": generateUserId(),
            "lastSessionDate": nil,
            "sessionHistory": newJArray(),
            "isFirstRun": true,
            "hasCompletedOnboarding": false,
            "hasUsedTrial": false,
            "preferences": %*{
              "theme": "light",
              "language": "en",
              "notifications": true
            },
            "tracking": %*{
              "lastCheck": nil,
              "checkCount": 0,
              "lastValidation": nil
            }
          }

        # Save new configuration
        writeFile(configPath, pretty(newConfig, 2))
        echo "✅ New configuration saved successfully\n"

        if "state.json" in configPath:
          echo "Account Details:"
          echo "User ID: ", newUserId
          echo "Device ID: ", newDeviceId
          echo "Email: ", userEmail
          echo "\nTrial period: 14 days"
          echo "Start date: ", trialStartDate.format("yyyy-MM-dd")
          echo "End date: ", trialEndDate.format("yyyy-MM-dd")
      except Exception as e:
        echo fmt"❌ Error processing {configPath}: {e.msg}"

    echo "\n🎉 Augment extension trial reset complete!"
    echo "\n⚠️ Important:"
    echo "1. Please restart your editor (VS Code or Cursor) for changes to take effect"
    echo "2. Create a new account when prompted"
    echo "3. The trial period will be active for 14 days"
    echo "4. Consider using a different network connection or VPN if issues persist"

    await waitForKeypress()
  except Exception as e:
    echo "❌ An error occurred: ", e.msg
    await waitForKeypress()

proc main() {.async.} =
  try:
    await resetAugmentTrial()
  except Exception as e:
    echo "\n❌ Program execution error: ", e.msg
    await waitForKeypress()

when isMainModule:
  waitFor main()

File: augment_reset.go
package main

import (
	"bufio"
	"crypto/rand"
	"encoding/hex"
	"encoding/json"
	"fmt"
	"io"
	"os"
	"os/exec"
	"path/filepath"
	"runtime"
	"strings"
	"time"
)

// Config structures for different JSON files
type StateConfig struct {
	DeviceID               string                 `json:"deviceId"`
	UserID                 string                 `json:"userId"`
	Email                  string                 `json:"email"`
	TrialStartDate         string                 `json:"trialStartDate"`
	TrialEndDate           string                 `json:"trialEndDate"`
	TrialDuration          int                    `json:"trialDuration"`
	TrialStatus            string                 `json:"trialStatus"`
	TrialExpired           bool                   `json:"trialExpired"`
	TrialRemainingDays     int                    `json:"trialRemainingDays"`
	TrialCount             int                    `json:"trialCount"`
	UsageCount             int                    `json:"usageCount"`
	TotalUsageCount        int                    `json:"totalUsageCount"`
	LastUsageDate          *string                `json:"lastUsageDate"`
	MessageCount           int                    `json:"messageCount"`
	TotalMessageCount      int                    `json:"totalMessageCount"`
	LastReset              string                 `json:"lastReset"`
	FirstRunDate           string                 `json:"firstRunDate"`
	LastRunDate            string                 `json:"lastRunDate"`
	SessionID              string                 `json:"sessionId"`
	LastSessionDate        *string                `json:"lastSessionDate"`
	SessionHistory         []interface{}          `json:"sessionHistory"`
	IsFirstRun             bool                   `json:"isFirstRun"`
	HasCompletedOnboarding bool                   `json:"hasCompletedOnboarding"`
	HasUsedTrial           bool                   `json:"hasUsedTrial"`
	Preferences            map[string]interface{} `json:"preferences"`
	Tracking               map[string]interface{} `json:"tracking"`
}

type SubscriptionConfig struct {
	Status         string        `json:"status"`
	Type           string        `json:"type"`
	StartDate      string        `json:"startDate"`
	EndDate        string        `json:"endDate"`
	IsActive       bool          `json:"isActive"`
	IsExpired      bool          `json:"isExpired"`
	RemainingDays  int           `json:"remainingDays"`
	TrialCount     int           `json:"trialCount"`
	LastTrialReset *string       `json:"lastTrialReset"`
	PreviousTrials []interface{} `json:"previousTrials"`
}

type AccountConfig struct {
	UserID        string        `json:"userId"`
	Email         string        `json:"email"`
	DeviceID      string        `json:"deviceId"`
	CreatedAt     string        `json:"createdAt"`
	LastLogin     string        `json:"lastLogin"`
	IsActive      bool          `json:"isActive"`
	TrialHistory  []interface{} `json:"trialHistory"`
	DeviceHistory []interface{} `json:"deviceHistory"`
}

// waitForKeypress waits for user input before exiting
func waitForKeypress() {
	if runtime.GOOS == "windows" && os.Getenv("TERM") == "" {
		fmt.Println("\nPress Enter to exit...")
	} else {
		fmt.Println("\nPress Enter to exit...")
	}

	reader := bufio.NewReader(os.Stdin)
	reader.ReadLine()
}

// isEditorRunning checks if VS Code or Cursor is currently running
func isEditorRunning() bool {
	var cmd *exec.Cmd

	switch runtime.GOOS {
	case "windows":
		cmd = exec.Command("tasklist", "/FI", "IMAGENAME eq Code.exe", "/FI", "IMAGENAME eq Cursor.exe")
	case "darwin":
		cmd = exec.Command("sh", "-c", "pgrep -x 'Code' || pgrep -x 'Cursor' || pgrep -x 'Code Helper' || pgrep -x 'Cursor Helper'")
	case "linux":
		cmd = exec.Command("sh", "-c", "pgrep -x 'code' || pgrep -x 'cursor' || pgrep -x 'Code' || pgrep -x 'Cursor'")
	default:
		return false
	}

	output, err := cmd.Output()
	if err != nil {
		return false
	}

	result := strings.ToLower(string(output))
	if runtime.GOOS == "windows" {
		return strings.Contains(result, "code.exe") || strings.Contains(result, "cursor.exe")
	}
	return len(strings.TrimSpace(string(output))) > 0
}

// killEditorProcess attempts to close running VS Code or Cursor processes
func killEditorProcess() bool {
	var cmd *exec.Cmd

	switch runtime.GOOS {
	case "windows":
		cmd = exec.Command("cmd", "/C", "taskkill /F /IM Code.exe /T & taskkill /F /IM Cursor.exe /T")
	case "darwin":
		cmd = exec.Command("sh", "-c", "pkill -9 'Code' & pkill -9 'Cursor'")
	case "linux":
		cmd = exec.Command("sh", "-c", "pkill -9 'code' & pkill -9 'cursor'")
	default:
		return false
	}

	err := cmd.Run()
	if err != nil {
		fmt.Printf("Error closing editors: %v\n", err)
		return false
	}

	// Wait for processes to close
	time.Sleep(1500 * time.Millisecond)

	return !isEditorRunning()
}

// formatTimestamp formats a time to a timestamp string
func formatTimestamp(t time.Time) string {
	return t.Format("20060102150405000")
}

// backupFile creates a backup of the specified file
func backupFile(filePath string) (string, error) {
	timestamp := formatTimestamp(time.Now())
	backupPath := fmt.Sprintf("%s.%s.bak", filePath, timestamp)

	sourceFile, err := os.Open(filePath)
	if err != nil {
		return "", fmt.Errorf("backup failed: %v", err)
	}
	defer sourceFile.Close()

	destFile, err := os.Create(backupPath)
	if err != nil {
		return "", fmt.Errorf("backup failed: %v", err)
	}
	defer destFile.Close()

	_, err = io.Copy(destFile, sourceFile)
	if err != nil {
		return "", fmt.Errorf("backup failed: %v", err)
	}

	return backupPath, nil
}

// getAugmentConfigPaths returns all possible Augment configuration file paths
func getAugmentConfigPaths() []string {
	var paths []string
	homeDir, _ := os.UserHomeDir()

	editors := []string{"Code", "Cursor"}
	configFiles := []string{"state.json", "subscription.json", "account.json"}
	cacheDirs := []string{"Cache", "CachedData"}

	switch runtime.GOOS {
	case "windows":
		appData := os.Getenv("APPDATA")
		localAppData := os.Getenv("LOCALAPPDATA")

		// Main configuration files
		for _, editor := range editors {
			for _, configFile := range configFiles {
				paths = append(paths, filepath.Join(appData, editor, "User", "globalStorage", "augment.augment", configFile))
			}
		}

		// Cache directories
		for _, editor := range editors {
			for _, cacheDir := range cacheDirs {
				paths = append(paths, filepath.Join(appData, editor, cacheDir, "augment.augment"))
			}
			paths = append(paths, filepath.Join(localAppData, editor, "User", "globalStorage", "augment.augment"))
		}

	case "darwin":
		// Main configuration files
		for _, editor := range editors {
			for _, configFile := range configFiles {
				paths = append(paths, filepath.Join(homeDir, "Library", "Application Support", editor, "User", "globalStorage", "augment.augment", configFile))
			}
		}

		// Cache directories
		for _, editor := range editors {
			paths = append(paths, filepath.Join(homeDir, "Library", "Caches", editor, "augment.augment"))
			paths = append(paths, filepath.Join(homeDir, "Library", "Application Support", editor, "Cache", "augment.augment"))
		}

	case "linux":
		// Main configuration files
		for _, editor := range editors {
			for _, configFile := range configFiles {
				paths = append(paths, filepath.Join(homeDir, ".config", editor, "User", "globalStorage", "augment.augment", configFile))
			}
		}

		// Cache directories
		for _, editor := range editors {
			paths = append(paths, filepath.Join(homeDir, ".cache", editor, "augment.augment"))
			paths = append(paths, filepath.Join(homeDir, ".config", editor, "Cache", "augment.augment"))
		}
	}

	return paths
}

// generateDeviceID generates a random device ID
func generateDeviceID() string {
	bytes := make([]byte, 32)
	rand.Read(bytes)
	return hex.EncodeToString(bytes)
}

// generateUserID generates a random user ID
func generateUserID() string {
	bytes := make([]byte, 16)
	rand.Read(bytes)
	return hex.EncodeToString(bytes)
}

// generateSessionID generates a random session ID
func generateSessionID() string {
	bytes := make([]byte, 16)
	rand.Read(bytes)
	return hex.EncodeToString(bytes)
}

// generateEmail generates a random email address
func generateEmail() string {
	bytes := make([]byte, 8)
	rand.Read(bytes)
	randomString := hex.EncodeToString(bytes)
	return fmt.Sprintf("user_%s@example.com", randomString)
}

// createStateConfig creates a new state configuration
func createStateConfig(deviceID, userID, email string, trialStart, trialEnd time.Time) *StateConfig {
	now := time.Now()
	return &StateConfig{
		DeviceID:               deviceID,
		UserID:                 userID,
		Email:                  email,
		TrialStartDate:         trialStart.Format(time.RFC3339),
		TrialEndDate:           trialEnd.Format(time.RFC3339),
		TrialDuration:          14,
		TrialStatus:            "active",
		TrialExpired:           false,
		TrialRemainingDays:     14,
		TrialCount:             0,
		UsageCount:             0,
		TotalUsageCount:        0,
		LastUsageDate:          nil,
		MessageCount:           0,
		TotalMessageCount:      0,
		LastReset:              now.Format(time.RFC3339),
		FirstRunDate:           now.Format(time.RFC3339),
		LastRunDate:            now.Format(time.RFC3339),
		SessionID:              generateSessionID(),
		LastSessionDate:        nil,
		SessionHistory:         []interface{}{},
		IsFirstRun:             true,
		HasCompletedOnboarding: false,
		HasUsedTrial:           false,
		Preferences: map[string]interface{}{
			"theme":         "light",
			"language":      "en",
			"notifications": true,
		},
		Tracking: map[string]interface{}{
			"lastCheck":      nil,
			"checkCount":     0,
			"lastValidation": nil,
		},
	}
}

// createSubscriptionConfig creates a new subscription configuration
func createSubscriptionConfig(trialStart, trialEnd time.Time) *SubscriptionConfig {
	return &SubscriptionConfig{
		Status:         "active",
		Type:           "trial",
		StartDate:      trialStart.Format(time.RFC3339),
		EndDate:        trialEnd.Format(time.RFC3339),
		IsActive:       true,
		IsExpired:      false,
		RemainingDays:  14,
		TrialCount:     0,
		LastTrialReset: nil,
		PreviousTrials: []interface{}{},
	}
}

// createAccountConfig creates a new account configuration
func createAccountConfig(deviceID, userID, email string, trialStart time.Time) *AccountConfig {
	return &AccountConfig{
		UserID:        userID,
		Email:         email,
		DeviceID:      deviceID,
		CreatedAt:     trialStart.Format(time.RFC3339),
		LastLogin:     trialStart.Format(time.RFC3339),
		IsActive:      true,
		TrialHistory:  []interface{}{},
		DeviceHistory: []interface{}{},
	}
}

// resetAugmentTrial performs the main reset operation
func resetAugmentTrial() error {
	fmt.Println("🔍 Checking for running editors...")
	if isEditorRunning() {
		fmt.Println("⚠️ VS Code or Cursor is running, attempting to close...")
		if killEditorProcess() {
			fmt.Println("✅ Editors have been closed\n")
		} else {
			fmt.Println("❌ Failed to close editors")
			return fmt.Errorf("failed to close editors")
		}
	}

	configPaths := getAugmentConfigPaths()
	fmt.Printf("📂 Found %d configuration paths\n", len(configPaths))

	// Generate new account data
	fmt.Println("🎲 Generating new account data...")
	newDeviceID := generateDeviceID()
	newUserID := generateUserID()
	userEmail := generateEmail()
	fmt.Println("✅ New account data generated successfully\n")

	// Calculate trial dates
	trialStartDate := time.Now()
	trialEndDate := trialStartDate.AddDate(0, 0, 14)

	var processedCount, errorCount int

	for _, configPath := range configPaths {
		fmt.Printf("\n🔄 Processing: %s\n", configPath)

		// Create directory if it doesn't exist
		if err := os.MkdirAll(filepath.Dir(configPath), 0755); err != nil {
			fmt.Printf("❌ Error creating directory: %v\n", err)
			errorCount++
			continue
		}

		// Backup existing config
		fmt.Println("💾 Backing up configuration...")
		if _, err := os.Stat(configPath); err == nil {
			if backupPath, err := backupFile(configPath); err == nil {
				fmt.Printf("✅ Configuration backup complete: %s\n", backupPath)
			} else {
				fmt.Printf("⚠️ Backup failed: %v\n", err)
			}
		} else {
			fmt.Println("ℹ️ No existing configuration to backup")
		}

		// Check if it's a directory and remove it
		if info, err := os.Stat(configPath); err == nil && info.IsDir() {
			if err := os.RemoveAll(configPath); err != nil {
				fmt.Printf("❌ Error removing directory: %v\n", err)
				errorCount++
				continue
			}
			fmt.Printf("✅ Removed directory: %s\n", configPath)
			processedCount++
			continue
		}

		// Create new configuration based on file type
		var configData interface{}
		if strings.Contains(configPath, "subscription.json") {
			configData = createSubscriptionConfig(trialStartDate, trialEndDate)
		} else if strings.Contains(configPath, "account.json") {
			configData = createAccountConfig(newDeviceID, newUserID, userEmail, trialStartDate)
		} else {
			configData = createStateConfig(newDeviceID, newUserID, userEmail, trialStartDate, trialEndDate)
		}

		// Save new configuration
		jsonData, err := json.MarshalIndent(configData, "", "  ")
		if err != nil {
			fmt.Printf("❌ Error marshaling JSON: %v\n", err)
			errorCount++
			continue
		}

		if err := os.WriteFile(configPath, jsonData, 0644); err != nil {
			fmt.Printf("❌ Error writing file: %v\n", err)
			errorCount++
			continue
		}

		fmt.Println("✅ New configuration saved successfully")
		processedCount++

		// Display account details for state.json files
		if strings.Contains(configPath, "state.json") {
			fmt.Println("\nAccount Details:")
			fmt.Printf("User ID: %s\n", newUserID)
			fmt.Printf("Device ID: %s\n", newDeviceID)
			fmt.Printf("Email: %s\n", userEmail)
			fmt.Println("\nTrial period: 14 days")
			fmt.Printf("Start date: %s\n", trialStartDate.Format("2006-01-02"))
			fmt.Printf("End date: %s\n", trialEndDate.Format("2006-01-02"))
		}
	}

	fmt.Printf("\n📊 Processing Summary:\n")
	fmt.Printf("Total paths: %d\n", len(configPaths))
	fmt.Printf("Successfully processed: %d\n", processedCount)
	fmt.Printf("Errors: %d\n", errorCount)

	fmt.Println("\n🎉 Augment extension trial reset complete!")
	fmt.Println("\n⚠️ Important:")
	fmt.Println("1. Please restart your editor (VS Code or Cursor) for changes to take effect")
	fmt.Println("2. Create a new account when prompted")
	fmt.Println("3. The trial period will be active for 14 days")
	fmt.Println("4. Consider using a different network connection or VPN if issues persist")

	return nil
}

// main function
func main() {
	fmt.Println("🚀 Augment Extension Trial Reset Tool")
	fmt.Println("====================================\n")

	if err := resetAugmentTrial(); err != nil {
		fmt.Printf("\n❌ An error occurred: %v\n", err)
		waitForKeypress()
		os.Exit(1)
	}

	waitForKeypress()
}
